apm-agent-plugins/apm-apache-httpclient/apm-apache-httpclient3-plugin/src/main/java/co/elastic/apm/agent/httpclient/v3/HttpClient3Instrumentation.java

```diff
@@ -82,54 +80,46 @@ public static class HttpClient3Advice {
         @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
         public static Object onEnter(@Advice.Argument(0) HttpMethod httpMethod,
                                      @Advice.FieldValue(value = "hostConfiguration") HostConfiguration hostConfiguration) {
-            final AbstractSpan<?> parent = TracerAwareInstrumentation.tracer.getActive();
-            if (parent == null) {
-                return null;
-            }
-
-            String host;
-            String uri;
-            String protocol;
-            int port;
-            URI httpClientURI = null;
-            try {
-                httpClientURI = httpMethod.getURI();
-                host = hostConfiguration.getHost();
-                port = hostConfiguration.getPort();
-                protocol = hostConfiguration.getProtocol().getScheme();
-                uri = httpClientURI.toString();
-            } catch (Exception e) {
+            Span<?> span = null;
+            final ElasticContext<?> parent = TracerAwareInstrumentation.tracer.currentContext();
+            if (parent.getAbstractSpan() != null) {
```

Comment:
[nit] I would still prefer the shorcut to keep the code a bit more linear (but that's just a preference, maybe let's discuss this during weekly meeting).